release/v1alpha1/types.go

```diff
@@ -0,0 +1,76 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PromotionTestResults encapsulates the inputs to produce a new set of promotion tests and
+// the status of the promotion tests
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type PromotionTestResults struct {
```

Comment:
@stevekuznetsov had a suggestion of ReleasePayload

Reply:
If I understand where @stevekuznetsov is heading with his work, the `PromotionTestResults` would be defined as part of the `Status` of a `ReleasePayload`.  The release-controller would be responsible for creating `ReleasePayload` objects, based on the configuration that is defined in a `ReleaseStream` object.

Reply:
> he release-controller would be responsible for creating `ReleasePayload` objects

Don't tell me, add a comment to your godoc :)

Also, "release-controller" is too vague.  if the individual control loops in it are not named, describe (in the godoc) at what stage this is created and what fields are set when it is created.

Reply:
rename this to ReleasePayload